Resync `svg` from WPT Upstream

Ahmad-S792 opened the pull request:
#### c3c6648720c7f0df32f6a38f228ccb124bc239f4
<pre>
Resync `svg` from WPT Upstream
<a href="https://bugs.webkit.org/show_bug.cgi?id=301489">https://bugs.webkit.org/show_bug.cgi?id=301489</a>
<a href="https://rdar.apple.com/163448269">rdar://163448269</a>

Reviewed by Anne van Kesteren and Brandon Stewart.

Upstream commit: <a href="https://github.com/web-platform-tests/wpt/commit/bc9587088f1e5621317f3db04b0f0d5f8e008b4f">https://github.com/web-platform-tests/wpt/commit/bc9587088f1e5621317f3db04b0f0d5f8e008b4f</a>

* LayoutTests/imported/w3c/resources/resource-files.json:
* LayoutTests/imported/w3c/web-platform-tests/svg/crashtests/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/svg/painting/reftests/marker-path-012.svg:
* LayoutTests/imported/w3c/web-platform-tests/svg/painting/reftests/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/svg/styling/required-properties-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/svg/styling/required-properties.svg:

Canonical link: <a href="https://commits.webkit.org/302445@main">https://commits.webkit.org/302445@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/9d8455e50c2599db4ff42699071dd5b2365b1330

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows | Apple Internal |
| ----- | ---------------------- | ------- |  ----- |  --------- | ------ |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/129108 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/159/builds/1367 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/39944 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/136487 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/80485 "Built successfully") | [✅ 🛠 ios-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/3327c40d-03c1-4b7c-be74-fd0afa915717) 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/155/builds/1299 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/156/builds/1244 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/98306 "Passed tests") | [❌ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/66177 "") | [❌ 🛠 mac-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/161fd5f3-44d6-46c7-81bc-10bacd521607) 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/132055 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/162/builds/1009 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/115653 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/78951 "Passed tests") | | [✅ 🛠 vision-apple](https://ews-bridge.webkit.apple.com/builds/sw/T-276/f2c6d2ed-bd8b-4d70-a499-64172ae81bd0/dd7629ef-2678-4813-9b90-10d320f1a5f6) 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/154/builds/932 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/33770 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/79766 "Built successfully") | | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/109379 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/34266 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/138961 "Built successfully") | | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/153/builds/1160 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/161/builds/1128 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/106842 "Passed tests") | | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/160/builds/1212 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/111989 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/106669 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/27156 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/164/builds/953 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/30513 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/53717 "Built successfully") | | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/158/builds/1233 "Built successfully") | | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/157/builds/1059 "Built successfully") | | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/163/builds/1105 "Built successfully") | | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/152/builds/1157 "Built successfully") | | | | 
<!--EWS-Status-Bubble-End-->

annevk reviewed (approved):
It seems like maybe there should be some coverage that text-space-collapse is not supported.

stwrt reviewed (approved):
marker-path-001 appears to be failing on Linux can you add that to the test expectations for Linux platforms.

Ahmad-S792 committed:
Resync `svg` from WPT Upstream
https://bugs.webkit.org/show_bug.cgi?id=301489
rdar://163448269

Reviewed by Anne van Kesteren and Brandon Stewart.

Upstream commit: https://github.com/web-platform-tests/wpt/commit/bc9587088f1e5621317f3db04b0f0d5f8e008b4f

* LayoutTests/imported/w3c/resources/resource-files.json:
* LayoutTests/imported/w3c/web-platform-tests/svg/crashtests/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/svg/painting/reftests/marker-path-012.svg:
* LayoutTests/imported/w3c/web-platform-tests/svg/painting/reftests/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/svg/styling/required-properties-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/svg/styling/required-properties.svg:

Canonical link: https://commits.webkit.org/302445@main
